src/libfetchers/fetchers.hh

```diff
@@ -213,8 +214,8 @@ struct PublicKey
     std::string type = "ssh-ed25519";
     std::string key;
 };
-NLOHMANN_DEFINE_TYPE_NON_INTRUSIVE_WITH_DEFAULT(PublicKey, type, key)
-
```

Comment:
Oh sorry, I didn't see this! With this we don't need `JSON_IMPL` because we already have a JSON impl.

Reply:
I thought we needed the new (custom) JSON Impl for the improved error messages (by using `getString` etc.)

Reply:
@haenoe That might be a great follow-up task (and you are welcome to submit it as a separate PR!), but it should not be part of this PR, which should be a "pure" refactor --- without any behavioral changes.